highlighters/main/main-highlighter.zsh

```diff
@@ -905,6 +904,12 @@ _zsh_highlight_main_highlighter_highlight_double_quote()
     (( k = j + 1 ))
     case "$arg[$i]" in
       '"') break;;
+      '`') saved_reply=($reply)
+           _zsh_highlight_main_highlighter_highlight_backtick $i
+           (( i = REPLY ))
+           reply=($saved_reply $reply)
```

Comment:
It might be cleaner to have the function append to some variable other than `reply`, e.g., `prospective_reply`, throughout and then do once `reply=( "${prospective_reply[@]}" )` at the end, to avoid workarounds such as this one.